processor/metricstransformprocessor/metrics_transform_processor.go

```diff
@@ -87,16 +91,50 @@ func (f internalFilterStrict) getSubexpNames() []string {
 	return nil
 }
 
+func (f internalFilterStrict) labelMatched(metric *metricspb.Metric) bool {
+	if len(f.matchLabels) == 0 {
+		return true
+	}
+
+	for key, value := range f.matchLabels {
+		keyFound := false
+
+		for idx, label := range metric.MetricDescriptor.LabelKeys {
+			if label.Key != key {
+				continue
+			}
+
+			keyFound = true
+			for _, timeseries := range metric.Timeseries {
+				if timeseries.LabelValues[idx].Value != value {
+					return false
+				}
+				break
```

Comment:
Because of this break, isn't only the first time series checked?

Reply:
Yes. That's I checked for. If the first timeseries matches the label, we will add the metric.

Reply:
Better to be explicit about that than use a loop then.

Reply:
Can you please rephrase for me? I don't think I understand what you meant.

Reply:
If you only want to check the first one can't you use `metric.Timeseries[0]`?

Reply:
Oh I see. Makes sense. Will update.

Reply:
Updated, thanks.